FrontEnd/src/ProfilePage/VendorsProfile.jsx

```diff
@@ -0,0 +1,53 @@
+import { useLocation } from "react-router-dom";
+import { useEffect, useState, useRef } from "react";
+import { fetchUserPosts } from "../UserAuthentication/FirestoreDB";
+import Modal from "../Modal";
+import { useTheme } from "../UseContext";
+import "./UserProfile.css";
+
+function VendorProfilePage() {
+  const { theme } = useTheme();
+  const location = useLocation();
+  const { post } = location.state ?? {};
+  const vendorID = post.vendorUID;
+  const [userPosts, setUserPosts] = useState([]);
+
+  useEffect(() => {
+    if (vendorID === null) return;
+    const unsubscribe = fetchUserPosts(vendorID, (postsData) => {
+      setUserPosts(postsData);
+    });
+    return () => unsubscribe && unsubscribe();
```

Comment:
Do you still need to check `unsubscribe == null` since you just defined it above?